### PR TITLE
Rework Database

### LINKED ISSUE #75: GitHub linking should work even if the user doesn't exist yet
GitHub linking currently requires the user to exist, so they can link their account.
This assumes someone has already gotten a coin on Discord, so they are a known user.
It should instead just create a new user with 0 coins and successfully link their GitHub account.

### PR DESCRIPTION
**Pull request by MahdiBM**
Cleaning up a bunch of things and reworking the database.

the new users table will have better column names, and the coin entries are also stored on another database so they don't slowdown getting raw user objects.

also resolves #75.

### CODE REVIEW

File: Sources/Penny/Services/CoinService/DefaultCoinService.swift [hunk 1]
```diff
@@ -69,64 +77,46 @@ actor DefaultCoinService: CoinService {
...
+        let url = "https://api.github.com/user/\(encodedID)"
+        logger.debug("Will make a request to get GitHub user name", metadata: [
+            "user": "\(user)",
+            "url": .string(url),
+        ])
+        var request = HTTPClientRequest(url: url)
+        request.headers = [
+            "Accept": "application/vnd.github+json",
+            "X-GitHub-Api-Version": "2022-11-28",
+            "User-Agent": "Penny/1.0.0 (https://github.com/vapor/penny-bot)"
```
Comment (gwynne) [inline Sources/Penny/Services/CoinService/DefaultCoinService.swift:104]: I do not at all love that this is hardcoded. (Nor for that matter the GitHub API version.)

File: Sources/SharedServices/DynamoUserRepository.swift [hunk 2]
```diff
@@ -14,113 +14,67 @@ struct DynamoUserRepository {
...
-    
-    init(db: DynamoDB, eventLoop: any EventLoop, logger: Logger) {
+    let tableName = "penny-user-table"
+    let discordIndex = "D-ID-GSI"
+    let githubIndex = "GH-ID-GSI"
+
+    init(db: DynamoDB, logger: Logger) {
         self.db = db
-        self.eventLoop = eventLoop
+        self.eventLoop = db.eventLoopGroup.any()
```
Comment (gwynne) [inline Sources/SharedServices/DynamoUserRepository.swift:28]: Bonus 😍😍😍 for knowing to use `.any()` instead of `.next()`!!! ❤️

**Review by gwynne — APPROVED** [review]
As Fabian loves to say, giphy

One small nit, but nbd.

File: Sources/Penny/Services/CoinService/DefaultCoinService.swift (hunk 1 above)
Comment (MahdiBM) [reply Sources/Penny/Services/CoinService/DefaultCoinService.swift:104]: What to do then?
It's not like it really matters to us.

**Comment by MahdiBM** [comment]
The test failures are due to migration stuff. Before migration stuff CI was all green.

**Comment by 0xTim** [comment]
I wish I'd had a chance to review this PR. Unfortunately this breaks many of the concepts of single table design we were taking advantage of before. It's fine now but it's going to make it more complex to evolve and scale in the future

**Comment by gwynne** [comment]
@0xTim Can you elaborate?

**Comment by MahdiBM** [comment]
I see Googling "DynamoDB singe-table design" comes up with some results.
I'll give them a read sometime soon to see what's the idea behind it, and if the idea seems as good/interesting to me as it apparently is to Tim.

**Pull request merged**